src/client/Microsoft.Identity.Client/ApiConfig/Executors/ConfidentialClientExecutor.cs

```diff
@@ -56,15 +57,16 @@ public ConfidentialClientExecutor(IServiceBundle serviceBundle, ConfidentialClie
                 requestContext,
                 _confidentialClientApplication.AppTokenCacheInternal);
 
-            if (clientParameters.AutoDetectRegion && requestContext.ServiceBundle.Config.AuthorityInfo.AuthorityType == AuthorityType.Adfs)
+            var autoDetectRegion = requestContext.ServiceBundle.Config.ExperimentalFeaturesEnabled;
```

Comment:
Using the experimental flag from api level. If it is selected for confidential client app, then set the autoDetectRegion flag to true here and everything else works as before.